Core/include/Acts/Propagator/DirectNavigator.hpp

```diff
@@ -195,15 +196,17 @@ class DirectNavigator {
       ACTS_VERBOSE("Current surface set to nullptr");
     }
 
-    // Reset the surface index
-    state.resetSurfaceIndex();
+    // Find initial index. Because the list of surfaces has the same order
+    // regardless the direction, we need to increment the surfaceIndex manually
+    // instead of calling nextSurface
     bool foundStartSurface = false;
+    state.surfaceIndex = 0;
     for (const Surface* surface : state.options.surfaces) {
       if (surface == state.currentSurface) {
         foundStartSurface = true;
         break;
       }
-      state.nextSurface();
+      state.surfaceIndex++;
```

Comment:
I believe it is important that `state.resetSurfaceIndex();` is called because it will set the index to `-1` or `size()` depending on the direction. `state.nextSurface();` will then increment or decrement depending on the direction.

What kind of problem are you facing here in the GSF?

Reply:
Didn't see that we are looping always in the same direction - that is a bug yes. Good catch 👍